Offices/Offices.API/Controllers/DoctorsController.cs

```diff
@@ -0,0 +1,179 @@
+﻿using AutoMapper;
+using FluentValidation;
+using Microsoft.AspNetCore.Mvc;
+using Offices.API.DTOs;
+using Offices.API.Extensions;
+using Offices.API.Validators;
+using Offices.Application.Models;
+using Offices.Application.Services.Abstractions;
+
+namespace Offices.API.Controllers
+{
+    /// <summary>
+    /// Controller for managing doctors.
+    /// </summary>
+    [Route("api/[controller]")]
+    [ApiController]
+    public class DoctorsController : ControllerBase
+    {
+        private readonly IDoctorService _doctorService;
+        private readonly ILogger<DoctorsController> _logger;
+        private readonly IMapper _mapper;
+
+        // Validators
+        private readonly IValidator<CreateDoctorDto> _createDoctorDtoValidator;
+        private readonly IValidator<UpdateDoctorDto> _updateDoctorDtoValidator;
+        private readonly IValidator<ObjectIdDto> _objectIdDtoValidator;
+
+        public DoctorsController(IDoctorService doctorService, ILogger<DoctorsController> logger, IMapper mapper,
+            IValidator<CreateDoctorDto> createDoctorDtoValidator,
+            IValidator<UpdateDoctorDto> updateDoctorDtoValidator,
+            IValidator<ObjectIdDto> objectIdDtoValidator)
+        {
+            _doctorService = doctorService;
+            _logger = logger;
+            _mapper = mapper;
+            _createDoctorDtoValidator = createDoctorDtoValidator;
+            _updateDoctorDtoValidator = updateDoctorDtoValidator;
+            _objectIdDtoValidator = objectIdDtoValidator;
+        }
+
+        /// <summary>
+        /// Gets the list of all doctors.
+        /// </summary>
+        /// <remarks>
+        /// Sample request:
+        ///
+        ///     GET /api/Doctors
+        ///
+        /// </remarks>
+        /// <returns>Returns a list of doctors objects.</returns>
+        /// <response code="200">Returns the list of doctors</response>
+        /// <response code="500">If there was an internal server error</response>
+        [HttpGet]
+        public async Task<IEnumerable<DoctorDto>> Get(CancellationToken cancellationToken)
+        {
+            var doctors = await _doctorService.GetAll(cancellationToken);
+            _logger.LogInformation("Requested doctors list");
+
+            var doctorDtos = _mapper.Map<IEnumerable<DoctorDto>>(doctors);
+            return doctorDtos;
+        }
+
+        /// <summary>
+        /// Gets an doctor by ID.
+        /// </summary>
+        /// <param name="id">The ID of the doctor to retrieve.</param>
+        /// <returns>Returns the doctor object.</returns>
+        /// <response code="200">If the doctor is found</response>
+        /// <response code="400">If the doctor is not found</response>
```

Comment:
404 should be returned everywhere when something wasn't found

Reply:
fixed